index.html

```diff
@@ -1494,12 +1532,24 @@ <h2>Core Vocabulary Definitions</h2>
 </p>
 </section>
 <section><h3><code>VersionInfo</code></h3><p>Carries version information about the TD instance. If required, additional version information such as firmware and hardware version (term definitions outside of the TD namespace) can be extended here.</p><table class="def"><thead><tr><th><a>Vocabulary term</a></th><th>Description</th><th>Assignment</th><th>Type</th></tr></thead><tbody><tr class="rfc2119-table-assertion" id="td-vocab-instance--VersionInfo"><td><code>instance</code></td><td>Provides a version identicator of this TD instance.</td><td>mandatory</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string"><code>string</code></a></td></tr></tbody></table></section>
-<section><h3><code>MultiLanguage</code></h3><p>Container to provide human-readable text in different languages using language tags described in [[BCP47]]. Each used language tags code MUST be decleared as vocabulary term member of this container (e.g., en, de, ja, zh-Hans, zh-Hant). See <a href="#titles-descriptions-serialization-json"></a> for example usages of this container in a Thing Description instance.</p>
+<section><h3><code>MultiLanguage</code></h3><p>Container to provide human-readable text in different languages using language tags described in [[BCP47]]. See <a href="#titles-descriptions-serialization-json"></a> for example usages of this container in a Thing Description instance.</p>
 <p>
+<span class="rfc2119-assertion" id="td-context-ns-multilanguage-language-tag">
+Each language tag code used MUST be declared as vocabulary term member of this container 
+(e.g., <code>en</code>, <code>de</code>, <code>ja</code>, <code>zh-Hans</code>, <code>zh-Hant</code>,
+<code>az-Latn</code>, <code>az-Arab</code>). </span>
+<span class="rfc2119-assertion" id="td-context-ns-multilanguage-text-direction-infer">
+The base text direction MUST be inferred from the language tag code.
+</span>
+<span class="rfc2119-assertion" id="td-context-ns-multilanguage-text-direction-script-subtag">
+If a language can be written in more than one script with different base directions 
+then a script subtag MUST be used so that the base direction can be inferred.
+</span>
 <span class="rfc2119-assertion" id="td-context-ns-multilanguage-content-negotiation">
-Note that in protocols such as HTTP that have an ability to negotiate content based on languages, 
-those terms (e.g. <code>description</code> and <code>title</code>) intended for carrying texts in a default language SHOULD be used instead of <code>MultiLanguage</code> container
-when the texts are served as a result such negotiation.
+In protocols such as HTTP that have an ability to negotiate content based on languages, 
+those terms (e.g. <code>description</code> and <code>title</code>) intended 
+for carrying texts in a default language SHOULD be used instead of <code>MultiLanguage</code> container
+when the texts are served as a result of such negotiation.
```

Comment:
Only works for dynamically serialized TDs...

Reply:
Actually, I don't think I changed this, it just got moved or the formatting got changed.   What you say is true, but I don't think it changes the assertion.

Reply:
I think there should be a comment after the assertions that informs that this does not mean that Things should be able to dynamically generate their TD. This might be hard for constrained devices.

Reply:
OK, I can add that comment.   SHOULD gives us some wiggle room.

Reply:
I added the following sentence immediately after this assertion (it is not itself an assertion, just a note): "Note however that constrained devices may choose to not support dynamically-generated TDs and do not need to support this feature."